src/utils/extract-date-parts-from-date-string.ts

```diff
@@ -7,10 +7,25 @@ export function extractDatePartsFromDateString(dateString: string) {
         throw new Error(`Date string is invalid, received "${dateString}"`)
     }
 
-    const [, yearStr, , monthStr, , dayStr] = parts
+    let yearStr, monthStr, dayStr, format
+
+    if (parts[1]) {
+        // Match for YYYY-MM-DD
+        yearStr = parts[1]
+        monthStr = parts[3]
+        dayStr = parts[5]
+        format = 'YYYY-MM-DD'
+    } else {
```

Comment:
can we also have a test (from the hook) to validate the effect of passing a completely different format (something like `2222-2222-2222` for example.

I see you have logic to fall back to today's date .. so you can have something like [this](https://github.com/dhis2/multi-calendar-dates/blob/88c6dc94b9b2b1c0f458db1a6093c9180954e944/src/hooks/useDatePicker.spec.tsx#L11) to ensure the test is not dependent on the actual date